src/main/java/org/apache/commons/text/numbers/DoubleFormat.java

```diff
@@ -0,0 +1,736 @@
+        }
+
+        /** Set the string containing the digit characters 0-9, in that order. The
+         * default value is the string {@code "0123456789"}.
+         * @param digits string containing the digit characters 0-9
+         * @return this instance
+         * @throws NullPointerException if the argument is null
+         * @throws IllegalArgumentException if the argument does not have a length of exactly 10
+         */
+        public Builder digits(final String digits) {
+            Objects.requireNonNull(digits, "Digits string cannot be null");
+            if (digits.length() != DEFAULT_DECIMAL_DIGITS.length()) {
+                throw new IllegalArgumentException("Digits string must contain exactly "
+                        + DEFAULT_DECIMAL_DIGITS.length() + " characters.");
+            }
+
+            this.digits = digits;
+            return this;
+        }
+
+        /** Set the flag determining whether or not a zero character is added in the fraction position
+         * when no fractional value is present. For example, if set to true, the number {@code 1} would
+         * be formatted as {@code "1.0"}. If false, it would be formatted as {@code "1"}. The default
+         * value is {@code true}.
+         * @param fractionPlaceholder if true, a zero character is placed in the fraction position when
+         *      no fractional value is present; if false, fractional digits are only included when needed
+         * @return this instance
+         */
+        public Builder includeFractionPlaceholder(final boolean fractionPlaceholder) {
+            this.fractionPlaceholder = fractionPlaceholder;
+            return this;
+        }
+
+        /** Set the character used as the minus sign.
+         * @param minusSign character to use as the minus sign
+         * @return this instance
+         */
+        public Builder minusSign(final char minusSign) {
+            this.minusSign = minusSign;
+            return this;
+        }
+
+        /** Set the decimal separator character, i.e., the character placed between the
+         * whole number and fractional portions of the formatted strings. The default value
+         * is {@code '.'}.
+         * @param decimalSeparator decimal separator character
+         * @return this instance
+         */
+        public Builder decimalSeparator(final char decimalSeparator) {
+            this.decimalSeparator = decimalSeparator;
+            return this;
+        }
+
+        /** Set the character used to separate groups of thousands. Default value is {@code ','}.
+         * @param groupingSeparator character used to separate groups of thousands
+         * @return this instance
+         * @see #groupThousands(boolean)
+         */
+        public Builder groupingSeparator(final char groupingSeparator) {
+            this.groupingSeparator = groupingSeparator;
+            return this;
+        }
+
+        /** If set to true, thousands will be grouped with the
```

Comment:
Nit-pick, but there are other `true`'s and `false`s (and I think `null`s?) that are formatted as plain text. Not a blocker for this PR though :+1: can be fixed later/someday.

Reply:
I think I've got them all now.